python/lsst/obs/hsc/hscMapper.py

```diff
@@ -127,7 +139,12 @@ def map(self, datasetType, dataId, write=False):
         """
         copyId = dataId.copy()
         copyId.pop("flags", None)
-        return super(HscMapper, self).map(datasetType, copyId, write=write)
+        location = super(HscMapper, self).map(datasetType, copyId, write=write)
+
+        if 'flags' in dataId:
+            location.getAdditionalData().set('flags', dataId['flags'])
+
+        return location
```

Comment:
Re-reading: I think the commit message is confusing here and that this probably isn't indicative of a bug in the `daf.persistence` Butler so much as a fix to the way `HscMapper` is interfacing with it. As such, the code is fine, but we should tweak the commit message.